presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -821,6 +829,11 @@ private boolean viewExists(ConnectorSession session, SchemaTableName viewName)
         return !getViews(session, viewName.toSchemaTablePrefix()).isEmpty();
     }
 
+    private boolean tableExists(ConnectorSession session, SchemaTableName tableName)
+    {
+        return listTables(session, tableName.getSchemaName()).contains(tableName);
```

Comment:
This will fetch all the tables. We can instead just do `getTableHandle()` which is designed to lookup a single table.